R/repair-column-names.R

```diff
@@ -0,0 +1,19 @@
+#' Check validity/format of a data.frame or tbl.
+#'
+#' Ensure the tbl or data.frame has legitimate and optionally unique
+#' column names.
+#' @param x tbl, tbl__df, or data.frame
+#' @param unique logical (default \code{TRUE}), force column names to
+#'     be unique (\code{TRUE}) or allow duplicates (\code{FALSE});
+#'     ignored if \code{check.names} is \code{FALSE}
+#' @return data.frame, with optionally different column names
+#' @export
+repair_names <- function(x, unique=TRUE) {
+    col_names <- colnames(x)
+    if (is.null(col_names)) {
+        col_names <- rep('', ncol(x))
```

Comment:
*`c("X", "X.1", "X.2")` versus `c("V1", "V2", "V3")`*: I had explicitly kept it that way, thinking simpler (fewer changes, more consistent with base R functions) is better.

I didn't realize you intended to reinvent a function to fix missing and duplicate headers. Your caveats are well-stated: I thought of doing it and argued myself out of it, precisely for the purpose of keeping simple code, using what has been working already, etc. I'll mull this over ...

*Empty names*: When you have a missing column, do you replace it with `V1` or the nth `V`? Are you concerned that skipping already-existing columns (`"^V[0-9]+$"`) can result in column names that are not sorted? I think I'd find that confusing. (Not entirely true ... I probably wouldn't let it get there in the first place :-)